File: data/Game.lua
--[[
   Xentu Code Example
   --------------------------
   An experiment to see how we can further clean-up the design of the lua side of xentu.
]]



-- called when the game first loads.
game.on("init", function()
	-- load a tilemap
	map = assets.load_tilemap("tiled-desert/desert.tmx")
	layer0 = map.get_layer(0)

	-- set the background.
	renderer.set_clear_color( Color.fromHex('#5f7988') )
	
    print("Hello from Lua world!")
	renderer.set_blend(true)
end)



-- the update event.
game.on("update", function()
	-- do nothing for now
end)



-- the drawing event.
game.on("draw", function()
    renderer.begin()

	renderer.draw_tilemap_layer(layer0)

    renderer.present()
end)

File: demos/01_tilemaps/game.py
print('Hello from python!')

# load resources.
tilemap1 = assets.load_tile_map_tmx("/assets/tilemap01.tmx", "/assets/")

# handle the update event.
def update_callback(dt):
	if (keyboard.key_clicked(const.KB_ESCAPE)):
		game.exit()

# handle the draw event.
def draw_callback(dt):
	renderer.clear()
	renderer.begin()
	renderer.draw_tile_layer(tilemap1, 0)
	renderer.draw_tile_layer(tilemap1, 1)
	renderer.draw_tile_layer(tilemap1, 2)
	renderer.present()


game.on("update", "update_callback")
game.on("draw", "draw_callback")